### PR TITLE
Bk 578 foh screens

### PR DESCRIPTION
**Pull request by jfplataroti**
Added pages `Ready` and `In Progress` for the FOH.

The page dimensions assumed that the screen size is the one provided in the mockups (1080x1920).

Base on the available height, total orders and expected item height (the item height without the need to resize them) it calculates if the space is big enough to render the items without resize them. If the items need to be resized it falls into three sizes:

- `large` -> ratio >= 0
- `medium` -> ratio < 0 and > 0.75
- 'small` -> ration < 0.75 and > 0.42
- `xsmall` -> not used at the moment.

# Ready Page

## Large

foh-ready-large-6

```
# Title
fontSize: '88px',
fontWeight: 800,
lineHeight: '96px',

# table header
fontSize: '32px',
fontWeight: 800,

# Username
fontSize: '60px',
fontWeight: 800,

# Order 
fontSize: '32px',
fontWeight: 600,

# Provider logo
height: '52px',
width: '52px',

# Cubby
height: '88px',
width: '140px',
fontSize: '56px',
```

## Medium 
(wasn’t in the mockups, was added to add an intermediate size)

foh-ready-medium-15

```
# Title
fontSize: '88px',
fontWeight: 800,
lineHeight: '96px',


# table header
fontSize: '32px',
fontWeight: 800,


# Username
fontSize: '48px',
fontWeight: 800,

# Order 
fontSize: '28px',
fontWeight: 600,


# Provider logo
height: '48px',
width: '48px',


# Cubby
height: '64px',
width: '100px',
fontSize: '32px',

```

## Small

fog-ready-small-33

```
# Title
fontSize: '48px',
fontWeight: 800,
lineHeight: '52px',

# table header
fontSize: '24px',
fontWeight: 800,

# Username
fontSize: '28px',
fontWeight: 800,

# Order 
fontSize: '28px',
fontWeight: 600,

# Provider logo
height: '38px',
width: '38px',

# Cubby
height: '40px',
width: '54px',
fontSize: '22px',

```

# In Progress Page

# In progress notes

## Large 
(only mockups for in progress)

foh-inprogress-large-8

```
# Title
fontSize: '88px',
fontWeight: 800,
lineHeight: '96px',

# table header
fontSize: '24px',
fontWeight: 800,

# Username
fontSize: '28px',
fontWeight: 800,

# Order 
fontSize: '28px',
fontWeight: 600,

# Provider logo
height: '52px',
width: '52px',
```

## Medium 
(not in mockups)

foh-inprogress-medium-32

```
# Title
fontSize: '88px',
fontWeight: 800,
lineHeight: '96px',

# table header
fontSize: '24px',
fontWeight: 800,

# Username
fontSize: '24px',
fontWeight: 800,

# Order 
fontSize: '24px',
fontWeight: 600,

# Provider logo
height: '48px',
width: '48px',
```

## Small

foh-inprogress-small-40

```
# Title
fontSize: '88px',
fontWeight: 800,
lineHeight: '96px',

# table header
fontSize: '24px',
fontWeight: 800,

# Username
fontSize: '20px',
fontWeight: 800,

# Order 
fontSize: '20px',
fontWeight: 600,

# Provider logo
height: '38px',
width: '38px',
```

### CODE REVIEW

File: common/config/rush/pnpm-lock.yaml [hunk 1]
```diff
@@ -11,7 +11,6 @@ dependencies:
   '@types/styled-components': 5.1.7
   '@typescript-eslint/eslint-plugin': 4.14.1_5aaa83948f53bc6d11a4ce27d9931983
   '@typescript-eslint/parser': 4.14.1_eslint@7.18.0+typescript@4.1.3
-  date-fns: 2.17.0
```
Comment (djoyal) [inline common/config/rush/pnpm-lock.yaml:14]: I got a bit confused looking at this 😅 

I think we might want to add this file to the `.gitignore`? It doesn't need to be part of this PR, just something we should think about I think (let me know if you disagree!)

File: packages/web/lib/theme.tsx [hunk 2]
```diff
@@ -67,10 +68,10 @@ const theme = {
   heading: {
     responsiveBreakpoint: undefined,
     extend: {
-      color: BRAND_COLORS.brand,
       letterSpacing: -0.02,
       textAlign: 'center',
     },
+    color: 'brand',
```
Comment (djoyal) [inline packages/web/lib/theme.tsx:74]: 👍

File: packages/web/pages/foh/[id]/inProgress.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                  size={size}
+                  value={row.deliveryProvider}
+                />
+              </Box>
+              <Box pad="small" justify="center" align="center">
+                <Heading
+                  level="5"
+                  style={inProgressOrderDimensions[size]}
+                  color="white">
+                  12345
```
Comment (djoyal) [inline packages/web/pages/foh/[id]/inProgress.tsx:108]: Should this be something like `row.id` or is this a placeholder?

File: packages/web/pages/foh/[id]/ready.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                  {row.username}
+                </Heading>
+              </Box>
+              <Box pad="small" justify="center" align="center">
+                <Heading
+                  alignSelf="center"
+                  margin={{ vertical: 'none' }}
+                  level={6}
+                  style={orderDimensions[size]}>
+                  12345
```
Comment (djoyal) [inline packages/web/pages/foh/[id]/ready.tsx:106]: Is this a placeholder value?

**Review by djoyal — COMMENTED** [review]
This looks great Jorge, nice work! 

There's a few small comments (mostly related to some placeholder values), but nothing major. 

Thanks for touching up the `theme.tsx` as well!

File: packages/web/pages/foh/[id]/inProgress.tsx (hunk 3 above)
Comment (jfplataroti) [reply packages/web/pages/foh/[id]/inProgress.tsx:108]: hahah yess, thanks for this one. 🤦‍♂️

File: packages/web/pages/foh/[id]/ready.tsx (hunk 4 above)
Comment (jfplataroti) [reply packages/web/pages/foh/[id]/ready.tsx:106]: yes heheh 🤦‍♂️

File: common/config/rush/pnpm-lock.yaml (hunk 1 above)
Comment (jfplataroti) [reply common/config/rush/pnpm-lock.yaml:14]: oh, maybe that was me resolving the conflicts. But yes good point, for now, we can move it to the .gitignore until the code is stable. 💯

**Review by djoyal — APPROVED** [review]
Thanks for correcting the placeholders, and for adding the `pnpm` file to the `.gitignore`!

**Pull request merged**